src/olympia/lib/crypto/tests.py

```diff
@@ -631,3 +631,13 @@ def test_sign_mail(self, mock_sign_file):
                 self.file_, settings.SIGNING_SERVER)
 
         assert 'expiration' in mail.outbox[0].message().as_string()
+
+
+def test_get_new_version_number():
+    for old, new in (
+            ['1.1', '1.1.1-signed'],
+            ['1.1.1-signed.1', '1.1.1-signed.1.1-signed'],
+            ['1.1.1-signed', '1.1.1-signed-2'],
+            ['1.1.1-signed-3', '1.1.1-signed-4'],
+            ['1.1.1-signed.1-signed-16', '1.1.1-signed.1-signed-17']):
+        assert tasks.get_new_version_number(old) == new
```

Comment:
This is a wonderful usecase for `pytest.mark.parametrize`

```python
@pytest.mark.parametrize(('old', 'new'), [
    ('1.1', '1.1.1-signed'),
    ('1.1.1-signed.1', '1.1.1-signed.1.1-signed'),
    ('1.1.1-signed', '1.1.1-signed-2'),
    ('1.1.1-signed-3', '1.1.1-signed-4'),
    ('1.1.1-signed.1-signed-16', '1.1.1-signed.1-signed-17'])
])
def test_get_new_version_number(old, new):
    assert tasks.get_new_version_number(old) == new
```

This then generates a separate  test-case for every variation and makes failures easier to digest and simpler to spot with better tracebacks / error messages since it's not hiding behind a `for` loop.